src/cpp/rtps/network/ExternalLocatorsProcessor.cpp

```diff
@@ -110,6 +110,27 @@ static void perform_add_external_locators(
     }
 }
 
+template<>
+void perform_add_external_locators<LocatorList>(
+        LocatorList& locators,
+        const ExternalLocators& external_locators)
+{
+    for (const auto& externality_item : external_locators)
+    {
+        // Only add locators with externality greater than 0
+        if (externality_item.first > 0)
+        {
+            for (const auto& cost_item : externality_item.second)
+            {
+                for (const auto& locator : cost_item.second)
+                {
+                    locators.push_back(locator);
+                }
+            }
+        }
```

Comment:
IDK whether we should have a logWarning on the else

Reply:
No, because `set_listening_locators` would have been previously called, and the input for this method would always have externality level 0 filled.